packages/medusa/src/strategies/batch-jobs/price-list/import.ts

```diff
@@ -0,0 +1,545 @@
+
+  static batchType = "price-list-import"
+
+  private processedCounter: Record<string, number> = {}
+
+  protected readonly featureFlagRouter_: FlagRouter
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly fileService_: IFileService
+
+  protected readonly regionService_: RegionService
+  protected readonly priceListService_: PriceListService
+  protected readonly productService_: ProductService
+  protected readonly batchJobService_: BatchJobService
+  protected readonly salesChannelService_: SalesChannelService
+  protected readonly productVariantService_: ProductVariantService
+  protected readonly shippingProfileService_: ShippingProfileService
+
+  protected readonly csvParser_: CsvParser<
+    PriceListImportCsvSchema,
+    Record<string, string>,
+    Record<string, string>
+  >
+
+  constructor({
+    batchJobService,
+    productVariantService,
+    priceListService,
+    regionService,
+    fileService,
+    manager,
+  }: InjectedProps) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.csvParser_ = new CsvParser(CSVSchema)
+
+    this.manager_ = manager
+    this.fileService_ = fileService
+    this.batchJobService_ = batchJobService
+    this.priceListService_ = priceListService
+    this.productVariantService_ = productVariantService
+    this.regionService_ = regionService
+  }
+
+  async buildTemplate(): Promise<string> {
+    throw new Error("Not implemented!")
+  }
+
+  /**
+   * Create a description of a row on which the error occurred and throw a Medusa error.
+   *
+   * @param row - Parsed CSV row data
+   * @param errorDescription - Concrete error
+   */
+  protected static throwDescriptiveError(
+    row: TParsedPriceListImportRowData,
+    errorDescription?: string
+  ): never {
+    const message = `Error while processing row with:
+      variant ID: ${row[PriceListRowKeys.VARIANT_ID]},
+      variant SKU: ${row[PriceListRowKeys.VARIANT_SKU]},
+      ${errorDescription}`
+
+    throw new MedusaError(MedusaError.Types.INVALID_DATA, message)
+  }
+
+  async prepareBatchJobForProcessing(
+    batchJob: CreateBatchJobInput,
+    reqContext: any
+  ): Promise<CreateBatchJobInput> {
+    if (!batchJob.context?.price_list_id) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "Price list id is required"
+      )
+    }
+
+    return batchJob
+  }
+
+  /**
+   * Generate instructions for update/create of products/variants from parsed CSV rows.
+   *
+   * @param csvData - An array of parsed CSV rows.
+   */
+  async getImportInstructions(
+    priceListId: string,
+    csvData: TParsedPriceListImportRowData[]
+  ): Promise<Record<OperationType, PriceListImportOperation[]>> {
+    const manager = this.transactionManager_ ?? this.manager_
+
+    // Validate that PriceList exists
+    await this.priceListService_.withTransaction(manager).retrieve(priceListId)
```

Comment:
suggestion: Would it make sense to move this to the `prepareBatchJobForProcessing` method? This way creation of the batch job would fail and the user would have information about the failure with the response.

Reply:
Excellent point!